### PR TITLE
in the case of wells do not consider their contribution in the comput…

### PR DESCRIPTION
**Pull request by alessiofumagalli**
…ation of the interpolated P0 velocity

### CODE REVIEW

File: src/porepy/numerics/vem/dual_elliptic.py [hunk 1]
```diff
@@ -43,9 +43,10 @@ def project_flux(gb, discr, flux, P0_flux, mortar_key="mortar_solution"):
             _, indices = np.unique(g.cell_faces.indices, return_index=True)
             sign = sps.diags(g.cell_faces.data[indices], 0)
 
-            for _, d_e in gb.edges_of_node(g):
+            for e, d_e in gb.edges_of_node(g):
                 g_m = d_e["mortar_grid"]
-                if g_m.dim == g.dim:
+                g1, _ = gb.nodes_of_edge(e)
+                if g_m.dim == g.dim or g1.well_num > -1:
```
Comment (IvarStefansson) [inline src/porepy/numerics/vem/dual_elliptic.py:49]: Should this only catch the cases when the lower-dimensional grid is a well? If not, would a check of the codimension be more appropriate?

Comment (alessiofumagalli) [reply src/porepy/numerics/vem/dual_elliptic.py:49]: so far my problem was with a well (second condition), but yes checking on the codimension includes this case. I'll make the change and propose it

**Comment by keileg** [comment]
Hi @alessiofumagalli: I have no idea which issue this PR resolves. Can you please provide background information here (this should really be a part of any PR), and document the code so that it is clear what happens?

**Comment by alessiofumagalli** [comment]
> Hi @alessiofumagalli: I have no idea which issue this PR resolves. Can you please provide background information here (this should really be a part of any PR), and document the code so that it is clear what happens?

You are right, @keileg . In the case with only fractures, the reconstruction of the velocity as P0 for each element is influenced by the mortar variable (flux) if a cell is connected to a 1-codimensional object. In the case of well, this does not apply and the contribution should be skip. In the general case for object of codimension >1 this contribution should not be considered. If you need any more details do not hesitate to ask.

**Comment by keileg** [comment]
Okay, that makes sense. 

BTW, is this part of the code covered by tests? The reason for asking is mainly #643, but more generally, since this code is not on the radar of any of the core developers, it is a weak point.

**Review by keileg — APPROVED** [review]

**Pull request merged**